docs/dev/operator-on-kind.md

```diff
@@ -0,0 +1,102 @@
+# Deploying the operator on a kind cluster
+
+The Infrastructure Operator can run on a kind cluster and will detect and alter its behavior to avoid relying on features only available in OpenShift.
+
+## How to deploy
+
+### Create a kind cluster
+
+The cluster should be created using a kind config which adds port mappings for ingress on http (port 80)
+
+For example:
+
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: InitConfiguration
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: "ingress-ready=true"
+  extraPortMappings:
+  - containerPort: 80
+    hostPort: 80
+    protocol: TCP
+  - containerPort: 443
+    hostPort: 443
+    protocol: TCP
+```
+
+```sh
+kind create cluster --config=<path/to/kind-config.yaml>
+```
+
+### Create required CRDs
+
+Some CRDs are not included in the Infrastructure Operator CSV and must be installed separately.
+For development these CRDs can be installed using copies in the assisted-service repo:
+
+```sh
+kubectl apply -f <assisted-service-repo>/hack/crds/*
+```
+
+This creates the ClusterDeployment, ClusterImageSet, and BareMetalHost CRDs.
+
+### Deploy the nginx ingress controller
+
+An ingress controller is needed to be able to create ingress resources.
+The nginx controller can be easily deployed with the following command:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
+```
+
+### Deploy the Infrastructure Operator and related CRDs from the local assisted-service repo
+
+```sh
+cd <assisted-service-repo>
+kustomize build config/default/ | oc apply -f -
+```
+
+This will result in all the required CRDs and RBAC being created as well as an operator deployment.
+
+### Create AgentServiceConfig
+
+An agent service config can now be created which should include the kubernetes-specific fields for ingress hosts and ingress class name:
+
+```yaml
+apiVersion: agent-install.openshift.io/v1beta1
+kind: AgentServiceConfig
+metadata:
+  name: agent
+spec:
+  ingressClassName: nginx
+  assistedServiceIngressHost: assisted-service.example.com
+  imageServiceIngressHost: image-service.example.com
```

Comment:
I think this isn't in sync with the CRD. Should be like this?

```yaml
ingress:
  className: nginx
  assistedServiceHostname: assisted-service.example.com
  imageServiceHostname: image-service.example.com
```

Reply:
Ah yea, let me update that. Thanks